wallet-new/src/Cardano/Wallet/WalletLayer/Kernel.hs

```diff
@@ -98,19 +95,10 @@ bracketPassiveWallet logFunction keystore rocksDB f = do
         invokeIO :: forall m'. MonadIO m' => Actions.WalletAction Blund -> m' ()
         invokeIO = liftIO . STM.atomically . invoke
 
-    -- The use of the unsafe constructor 'UnsafeRawResolvedBlock' is justified
-    -- by the invariants established in the 'Blund'.
-    blundToResolvedBlock :: (Core.SlotId -> IO Core.Timestamp) -> Blund -> IO (Maybe ResolvedBlock)
-    blundToResolvedBlock getTimeBySlot (b,u) = do
-        case b of
-            Left _ ->  return Nothing
-            Right mainBlock ->  do
-                let slot = mainBlock ^. mainBlockSlot
-                time <- getTimeBySlot slot
-                return . Just $ fromRawResolvedBlock
-                    (UnsafeRawResolvedBlock mainBlock spentOutputs' time)
-        where
-            spentOutputs' = map (map fromJust) $ undoTx u
+        getWal :: WalletId -> n (Either GetWalletError Wallet)
+        getWal wId = ro (Wallets.getWallet wId) >>= \case
+            Right v1wal -> Right <$> Wallets.updateSyncState w v1wal
+            err         -> return err
```

Comment:
I agree that it makes for sense to just set the sync state in `getWallet`. However, `getWallet` is pure and `updateSyncState` isn't, because it has to peek into the `MVar` that holds all of the restoration progress data. I guess I could keep `getWallets` pure and explicitly pass in the `WalletRestorationInfo`. Then this `getWal` will just have to read the `MVar` and pass the result over to `getWallet`.

`getWallets` would need a related change, but I'm less clear about what the right thing to do there is.